### PR TITLE
[ASIM Parsers] Add GitHub action that generates ARM templates from KQ…

### PR DESCRIPTION
**Pull request by sivanguetta**
…L YAML files

   Required items, please complete
   
   Change(s):
   - See guidance below

   Reason for Change(s):
   - See guidance below

   Version Updated:
   - Required only for Detections/Analytic Rule templates
   - See guidance below

   Testing Completed:
   - See guidance below

   Checked that the validations are passing and have addressed any issues that are present:
   - See guidance below

# Guidance <- remove section before submitting
-----------------------------------------------------------------------------------------------------------
## Before submitting this PR please ensure that you have read the following sections and filled out the changes, reason for change and testing complete sections:

Thank you for your contribution to the Microsoft Sentinel Github repo.

> Details of the code changes in your submitted PR.  Providing descriptions for pull requests ensures there is context to changes being made and greatly enhances the code review process.  Providing associated Issues that this resolves also easily connects the reason.
   
   Change(s):
   - Updated syntax for XYZ.yaml

   Reason for Change(s):
   - New schema used for XYZ.yaml
   - Resolves ISSUE #1234

   Version updated:
   - Yes
   - Detections/Analytic Rule templates are required to have the version updated

> The code should have been tested in a Microsoft Sentinel environment that does not have any custom parsers, functions or tables, so that you validate no incorrect syntax and execution functions properly.  If your submission requires a custom parser or function, it must be submitted with the PR.

   Testing Completed:
   - Yes/No/Need Help

_Note: If updating a detection, you must update the version field._

> Before the submission has been made, please look at running the KQL and Yaml Validation Checks locally.
> https://github.com/Azure/Azure-Sentinel#run-kql-validation-locally

   Checked that the validations are passing and have addressed any issues that are present:
   - Yes/No/Need Help
   
   _Note: Let us know if you have tried fixing the validation error and need help._

> References: 
> - [Guidance for Detection checks](https://github.com/Azure/Azure-Sentinel#pull-request-detection-template-structure-validation-check)
> - [General contribution guidance](https://github.com/Azure/Azure-Sentinel/wiki#what-can-you-contribute-and-how-can-you-create-contributions)
> - [PR validation troubleshooting](https://github.com/Azure/Azure-Sentinel#pull-request)

-----------------------------------------------------------------------------------------------------------

### CODE REVIEW

File: .github/workflows/convertKqlFuncYamlToArmTemplate.yaml [hunk 1]
```diff
@@ -0,0 +1,52 @@
+name: Convert Kql function yaml to ARM template
```
Comment (ori-licht) [inline .github/workflows/convertKqlFuncYamlToArmTemplate.yaml:1]: Can you add a description to this Yaml?
explaining what it does?

File: .script/kqlFuncYaml2Arm.sh [hunk 2]
```diff
@@ -0,0 +1,19 @@
+#!/bin/bash
+
+failed=0
+
+parsersSchemas=(ASimDns ASimNetworkSession ASimWebSession)
```
[the remark this answers is not in the document]
Comment (sivanguetta) [reply .script/kqlFuncYaml2Arm.sh:5]: > so for new schema we need to add its name to this list? if so maybe we should add a comment?

yes, I will add a comment

File: .github/workflows/convertKqlFuncYamlToArmTemplate.yaml [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+on:
+  pull_request_target:
+    types: [opened, edited, reopened, synchronize, closed, labeled]
```
Comment (oshezaf) [inline .github/workflows/convertKqlFuncYamlToArmTemplate.yaml:4]: Didn't you mention that you can't run this on a closed trigger type, which is when the branch is merged?

File: .github/workflows/convertKqlFuncYamlToArmTemplate.yaml [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    paths:
+       - 'Parsers/ASimDns/Parsers/**'
+       - 'Parsers/ASimNetworkSession/Parsers/**'
+       - 'Parsers/ASimWebSession/Parsers/**'
+jobs:
+  kqlFuncYaml2Arm:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
```
Comment (oshezaf) [inline .github/workflows/convertKqlFuncYamlToArmTemplate.yaml:13]: Doesn't the first set require a name?

**Review by oshezaf — COMMENTED** [review]
Just a few non-blocking comments.

File: .github/workflows/convertKqlFuncYamlToArmTemplate.yaml (hunk 4 above)
Comment (sivanguetta) [reply .github/workflows/convertKqlFuncYamlToArmTemplate.yaml:13]: Name property is not optional, but it is good for documentation and logs.
Adding name

File: .script/kqlFuncYaml2Arm.sh [hunk 5]
```diff
@@ -0,0 +1,21 @@
+#!/bin/bash
+
+failed=0
+
+# The KqlFuncYaml2Arm script generates deployable ARM templates from KQL function YAML files.
+# Currently, the script only runs on the parsers listed below.
```
Comment (ori-licht) [inline .script/kqlFuncYaml2Arm.sh:6]: nit:
on the Schemas listed below? instead of parsers?

**Pull request merged**